docs/src/main/asciidoc/reactive-routes.adoc

```diff
@@ -522,6 +522,171 @@ public class MyFilters {
+                type: string
+----
+
+Also see link:https://quarkus.io/guides/openapi-swaggerui[the OpenAPI Guide]
+
+=== Adding MicroProfile OpenAPI Annotations
+
+You can use link:https://github.com/eclipse/microprofile-open-api[MicroProfile OpenAPI] to better document your schema, 
+example, adding header info, or specifying the return type on `void` methods might be usefull :
+
+[source, java]
+----
+@OpenAPIDefinition(<1>
+    info = @Info(
+        title="Greeting API",
+        version = "1.0.1",
+        contact = @Contact(
+            name = "Greeting API Support",
+            url = "http://exampleurl.com/contact",
+            email = "techsupport@example.com"),
+        license = @License(
+            name = "Apache 2.0",
+            url = "http://www.apache.org/licenses/LICENSE-2.0.html"))
+)
+@ApplicationScoped
+public class MyDeclarativeRoutes {
+
+    // neither path nor regex is set - match a path derived from the method name
+    @Route(methods = HttpMethod.GET)
+    @APIResponse(responseCode="200", 
+            description="Say hello", 
+            content=@Content(mediaType="application/json", schema=@Schema(type=SchemaType.STRING)))<2>
+    void hello(RoutingContext rc) {
+        rc.response().end("hello");
+    }
+    
+    @Route(path = "/world") 
+    String helloWorld() {
+        return "Hello world!";
+    }
+
+    @Route(path = "/greetings", methods = HttpMethod.GET)
+    @APIResponse(responseCode="200", 
+            description="Greeting", 
+            content=@Content(mediaType="application/json", schema=@Schema(type=SchemaType.STRING)))
+    void greetings(RoutingExchange ex) {
+        ex.ok("hello " + ex.getParam("name").orElse("world"));
+    }
+}
+----
+
+<1> Header information about your API.
+<2> Defining the response
+
+This will generate this OpenAPI schema:
+
+[source, yaml]
+----
+---
+openapi: 3.0.3
+info:
+  title: Greeting API
+  contact:
+    name: Greeting API Support
+    url: http://exampleurl.com/contact
+    email: techsupport@example.com
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.1
+paths:
+  /greetings:
+    get:
+      responses:
+        "200":
+          description: Greeting
+          content:
+            application/json:
+              schema:
+                type: string
+  /hello:
+    get:
+      responses:
+        "200":
+          description: Say hello
+          content:
+            application/json:
+              schema:
+                type: string
+  /world:
+    get:
+      responses:
+        "200":
+          description: OK
+          content:
+            '*/*':
+              schema:
+                type: string
+----
+
+=== Using Swagger UI
+
+Swagger UI is included by default when running in `Dev` or `Test` mode, and can optionally added to `Prod` mode.
+See link:https://quarkus.io/guides/openapi-swaggerui#use-swagger-ui-for-development[the Swagger UI] Guide for more details.
```

Comment:
Is this the correct way to navigate to a section in another document? CC @gsmet

The URL part is probably pretty stable but I think that `<<openapi-swaggerui.adoc#use-swagger-ui-for-development,the Swagger UI>>` should work as well.

Reply:
Ok let me change that.

Reply:
mmm, the rest of the document use full URLs (see line 78 - 87) - should I change all of them. Or stay as is to be consistent ?

Reply:
Forget that. Those are different domains. I am changing.